### ISSUE TITLE
[Bug] Certain hyperparameter combinations can trigger low accuracy outcome of SMAP classifier

### ISSUE THREAD

**Issue by marisn** [OSGeo/grass#2970]
During a hyper-parameter optimization run with certain i.gensigset maxsig values a really low accuracy classification outcome was observed. The runs resulted in overall accuracy of ~20% meanwhile adjusting maxsig by one gave accuracy around 75% (as expected).

The root cause of accuracy drop needs to be determined. Just taking a note to not forget to examine this potential bug in the future.
